torch/autograd/saved_variable_default_hooks.py

```diff
@@ -1,7 +1,35 @@
 import torch
+from typing import Any
 
 def set_saved_tensors_default_hooks(pack_hook, unpack_hook):
     torch._C._autograd._register_default_hooks(pack_hook, unpack_hook)
 
 def reset_saved_tensors_default_hooks():
     torch._C._autograd._reset_default_hooks()
+
+class save_on_cpu(object):
+    def __init__(self, pin_memory=False):
+        def pack_to_cpu(tensor):
+            if not pin_memory:
+                return (tensor.device, tensor.cpu())
+
+            storage = torch.empty(
+                tensor.size(),
+                dtype=tensor.dtype,
+                layout=tensor.layout,
+                pin_memory=(torch.cuda.is_available() and not tensor.is_sparse))
+            storage.copy_(tensor)
```

Comment:
Why can't we do `storage = tensor.to(device='cpu', pin_memory=True)`?

Also nit: we should rename storage into something else; storage can be confused with `torch.Storage`

Reply:
I don't think that `pin_memory` is an acceptable argument of `torch.tensor.to`

Ok for `storage` rename.

Reply:
> I don't think that `pin_memory` is an acceptable argument of `torch.tensor.to`

Good point, thanks for the clarification

Reply:
would something like `storage = tensor.to("cpu", non_blocking=True).pin_memory()` work?

Reply:
That would mean 2 copies